src/storage/QueryEdgeKeysProcessor.cpp

```diff
@@ -0,0 +1,52 @@
+/* Copyright (c) 2019 vesoft inc. All rights reserved.
+ *
+ * This source code is licensed under Apache 2.0 License,
+ * attached with Common Clause Condition 1.0, found in the LICENSES directory.
+ */
+
+#include "base/NebulaKeyUtils.h"
+#include "storage/QueryEdgeKeysProcessor.h"
+
+
+namespace nebula {
+namespace storage {
+
+void QueryEdgeKeysProcessor::process(const cpp2::EdgeKeyRequest& req) {
+    auto spaceId = req.get_space_id();
+    auto vId = req.get_vid();
+    auto partId = req.get_part_id();
+    CHECK_NOTNULL(kvstore_);
+
+    std::vector<cpp2::EdgeKey> edges;
+    auto prefix = NebulaKeyUtils::prefix(partId, vId);
+    std::unique_ptr<kvstore::KVIterator> iter;
+    auto ret = this->kvstore_->prefix(spaceId, partId, prefix, &iter);
+    if (ret != kvstore::ResultCode::SUCCEEDED) {
+        VLOG(3) << "Error! ret = " << static_cast<int32_t>(ret) << ", spaceId = " << spaceId
+                << ", partId =  " << partId << ", vertexId = " << vId;
+        this->pushResultCode(this->to(ret), partId);
+        this->onFinished();
+        return;
+    }
+    while (iter->valid()) {
+        auto key = iter->key();
+        if (NebulaKeyUtils::isEdge(key)) {
+            auto src = NebulaKeyUtils::getSrcId(key);
+            auto dst = NebulaKeyUtils::getDstId(key);
+            auto edgeType = NebulaKeyUtils::getEdgeType(key);
+            auto rank = NebulaKeyUtils::getRank(key);
+            auto outEdge = cpp2::EdgeKey(apache::thrift::FragileConstructor::FRAGILE,
+                                        src, edgeType, rank, dst);
+            auto inEdge = cpp2::EdgeKey(apache::thrift::FragileConstructor::FRAGILE,
+                                        dst, -edgeType, rank, src);
+            edges.emplace_back(std::move(outEdge));
+            edges.emplace_back(std::move(inEdge));
```

Comment:
because all the fields of inEdge and outEdge is same else edgeType, so maybe there is no need return two edge, so cult down the flow of network

Reply:
Good idea